near-sdk-macros/src/core_impl/code_generator/impl_item_method_info.rs

```diff
@@ -118,14 +118,22 @@ impl ImplItemMethodInfo {
             let decomposition = self.attr_signature_info.decomposition_pattern();
             let serializer_invocation = match self.attr_signature_info.input_serializer {
                 SerializerType::JSON => quote! {
-                    ::near_sdk::serde_json::from_slice(
-                        &::near_sdk::env::input().expect("Expected input since method has arguments.")
-                    ).expect("Failed to deserialize input from JSON.")
+                    match ::near_sdk::env::input() {
+                        Some(input) => match ::near_sdk::serde_json::from_slice(&input) {
+                            Ok(deserialized) => deserialized,
+                            Err(_) => near_sdk::env::panic_str("Failed to deserialize input from JSON.")
+                        },
+                        None => near_sdk::env::panic_str("Expected input since method has arguments.")
```

Comment:
Please, use fully-qualified syntax here (add `::` in the beginning) since this code is generated into the user code and we want to avoid an unexpected aliasing:

```suggestion
                        None => ::near_sdk::env::panic_str("Expected input since method has arguments.")
```

P.S. Apply this change in other places where applicable (in near-sdk-marcos)